src/coorddesc.jl

```diff
@@ -0,0 +1,153 @@
+# Coordinate descent method, translated from the Python/Cython implementation
+#  in scikit-learn and modified to comply with the interfaces of the NMF package
+
+# Original files
+# https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/decomposition/nmf.py
+# https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/decomposition/cdnmf_fast.pyx
+
+# Original implementation authors:
+# Vlad Niculae
+# Lars Buitinck
+# Mathieu Blondel <mathieu@mblondel.org>
+# Tom Dupre la Tour
+
+# Original license: BSD 3 clause
+
+# Julia translation: Vilim Štih
+
+# Reference: Cichocki, Andrzej, and P. H. A. N. Anh-Huy. "Fast local algorithms for
+#  large scale nonnegative matrix and tensor factorizations."
+#  IEICE transactions on fundamentals of electronics, communications and
+#  computer sciences 92.3: 708-721, 2009.
+
+
+mutable struct TransposedArray{T} <: AbstractArray{T,2}
```

Comment:
Is this actually necessary? Starting with Julia 0.7, `transpose` and `adjoint` produce lazy wrappers of the underlying matrix, which I think should fulfill this purpose.

Reply:
Thanks for catching this. I was actually using permuteddimsarray, which has now been updated to use the lazy transpose.